node/logger/logger.go

```diff
@@ -207,3 +208,57 @@ func Panicf(format string, args ...interface{}) {
 func GetLevel() string {
 	return getInstance().GetLevel().String()
 }
+
+func getStringFromCall(request interface{}, methodName string) string {
+	var returnString string
+	method := reflect.ValueOf(request).MethodByName(methodName)
+	if method.IsValid() {
+		returnValue := method.Call([]reflect.Value{})[0]
+		returnString, _ = returnValue.Interface().(string)
+	}
+	return returnString
+}
+
+func setLoggerVolumeId(request interface{}) {
+	volumeId := getStringFromCall(request, "GetVolumeId")
+	goid_info.SetAdditionalIDInfo(volumeId)
+}
+
+func getFuncName() string {
+	var funcName string
+	pc, _, _, ok := runtime.Caller(2)
```

Comment:
can you change `pc` to `programCounter` or something more indicative?

Reply:
done